crates/e2e/tests/e2e/partially_fillable_observed_score.rs

```diff
@@ -145,17 +145,24 @@ async fn test(web3: Web3) {
         .await
         .unwrap();
 
-    let competition_0 = services
-        .get_solver_competition(trades[0].tx_hash.unwrap())
-        .await
-        .unwrap();
-    let competition_1 = services
-        .get_solver_competition(trades[1].tx_hash.unwrap())
-        .await
-        .unwrap();
-
+    let mut competitions: Vec<SolverCompetitionAPI> = vec![
+        services
+            .get_solver_competition(trades[0].tx_hash.unwrap())
+            .await
+            .unwrap(),
+        services
+            .get_solver_competition(trades[1].tx_hash.unwrap())
+            .await
+            .unwrap(),
+    ];
+    competitions.sort_by(|a, b| a.auction_id.cmp(&b.auction_id));
+    let competition_0 = competitions[0].clone();
+    let competition_1 = competitions[1].clone();
```

Comment:
Yes, the issue was that the competition_0 and competition_1 had no order relation with the auction_id.
So instead of doing
if competition_0. auction_id >  competition_1.auction_id {
    assert(latest, competition_1);
} else {
    assert(latest, competition_0);
}

I would rather sort them by the auction_id

Reply:
Ahh, right. I remembered the test setup differently.
Maybe it makes sense to patch `/api/v1/trades` such that it returns trades ordered by block that they happened in in a different PR.